pkg/virt-controller/services/template.go

```diff
@@ -499,6 +499,25 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 			}
 			sidecarVolumes = append(sidecarVolumes, vol)
 		}
+		if requestedHookSidecar.PVC != nil {
+			volumeSource := k8sv1.VolumeSource{
+				PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+					ClaimName: requestedHookSidecar.PVC.Name,
+				},
+			}
+			vol := k8sv1.Volume{
+				Name:         requestedHookSidecar.PVC.Name,
+				VolumeSource: volumeSource,
+			}
+			sidecarVolumes = append(sidecarVolumes, vol)
+			if requestedHookSidecar.PVC.SharedComputePath != "" {
+				containers[0].VolumeMounts = append(containers[0].VolumeMounts,
+					k8sv1.VolumeMount{
+						Name:      requestedHookSidecar.PVC.Name,
+						MountPath: requestedHookSidecar.PVC.SharedComputePath,
+					})
+			}
```

Comment:
Could this be a condition of its own at the same level as `if requestedHookSidecar.PVC != nil`. Mainly to reduce the cyclomatic complexity.

Reply:
It could reduce the nesting if but I need to recheck if the PVC is nil and it would add another condition to the if. Hence, I'm not sure if this is more readable

Reply:
Totally missed the possibility of `nil` here. 🤦🏽‍♂️